.github/workflows/run_tests.yaml

```diff
@@ -0,0 +1,73 @@
+name: Run rocks tests
+
+on:
+  workflow_call:
+    inputs:
+      rock-metas:
+        description: List of maps featuring the built {name, version, path, arch, image}
+        type: string
+        default: "[]"
+
+jobs:
+  run-tests:
+    if: ${{ inputs.rock-metas != '[]' }}
+    name: Run tests
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Maximize build space
+        uses: AdityaGarg8/remove-unwanted-software@v4.1
+        with:
+          remove-android: 'true'
+          remove-haskell: 'true'
+          remove-codeql: 'true'
+          remove-swapfile: 'true'
+      - name: Checkout repository
+        uses: actions/checkout@v4.1.1
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.8'
+      - name: Setup environment
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y jq
+          pip install tox
+      - name: Install multipass
+        run: |
+          set -x
+          sudo snap install multipass
+
+          # We need to authenticate into multipass, since we're not running as root.
+          # https://github.com/canonical/multipass/issues/1437#issuecomment-1435357872
+          sudo multipass set local.passphrase=Passw0rd
+
+          # If we auth too soon, we might get this error:
+          # authenticate failed: cannot connect to the multipass socket
+          sleep 3
+          multipass auth Passw0rd
+      - name: Set rock image environment variables
+        run: |
+          # NOTE(aznashwan): these env var names are maintained for backwards
+          # compatibility, but new tests should ideally use the utility functions
+          # from the shared testing harnesss repository to directly read the
+          # values from the below-set BUILT_ROCKS_METADATA environment variable.
+          # https://github.com/canonical/k8s-test-harness/blob/main/k8s_test_harness/util/env_util.py
+          echo '${{ inputs.rock-metas }}' | jq -c '.[] | select(.arch == "amd64") ' | while read i; do
```

Comment:
why do we maintain backwards compatibility for a new ROCK?
Also, this is an integration test that we own, why do we need to maintain backward compatibility?

Reply:
This was copied over from the original ``canonical/k8s-workflows/.github/workflows/run_tests.yaml@main`` action as is. I guess we can remove this from this repo specifically.